### PR TITLE
Add sublists

### PR DESCRIPTION
**Pull request by nayakrujul**

### CODE REVIEW

**Comment by nayakrujul** [comment]
Yeah, I'm not too sure if the `prefixes` and `suffixes` functions can handle infinite lists. If it fails that will probably be the reason.

File: shared/src/vyxal/ListHelpers.scala [hunk 1]
```diff
@@ -467,6 +467,9 @@ object ListHelpers:
 
   def prefixes(iterable: VList): Seq[VList] = iterable.inits.toSeq.reverse.tail
 
+  def suffixes(iterable: VList): Seq[VList] =
+    prefixes(iterable.reverse).reverse.map(a => a.reverse)
```
[the remark this answers is not in the document]
Comment (Lyxal) [reply shared/src/vyxal/ListHelpers.scala:471]: I think that's acceptable for suffixes. After all, how can suffixes work on infinite lists?

File: shared/src/vyxal/ListHelpers.scala [hunk 2]
```diff
@@ -668,4 +671,9 @@ object ListHelpers:
       makeIterable(lst).sortBy((a: VAny) => ListHelpers.makeIterable(a).length)
     )
 
+  def flattenOnce(lst: VList)(using Context): VList =
```
[the remark this answers is not in the document]
Comment (nayakrujul) [reply shared/src/vyxal/ListHelpers.scala:674]: I wanted to flatten `list[list[list[int]]]` into `list[list[int]]`, but `flatten` gives me `list[int]` (completely flattens it)

File: shared/src/vyxal/ListHelpers.scala [hunk 3]
```diff
@@ -467,6 +467,9 @@ object ListHelpers:
 
   def prefixes(iterable: VList): Seq[VList] = iterable.inits.toSeq.reverse.tail
```
Comment (Lyxal) [inline shared/src/vyxal/ListHelpers.scala:468]: So uh suffixes may be fine to not work on infinite lists,

but turns out prefixes doesn't work on inf lists either.

And it should

**Review by Lyxal — CHANGES_REQUESTED** [review]

File: shared/src/vyxal/ListHelpers.scala (hunk 3 above)
Comment (nayakrujul) [reply shared/src/vyxal/ListHelpers.scala:468]: Not my code - ask @ysthakur

**Review by Lyxal — CHANGES_REQUESTED** [review]
One second I'm adding tests for inf lists

**Comment by Lyxal** [comment]
> Suffixes should work on infinite lists too, it's actually super easy: just keep doing tail

Goofy how you gonna know the last item of an infinite list.

**Comment by nayakrujul** [comment]
@Lyxal suffixes works on infinite lists in v2, just like how @ysthakur said. But, I don't know how useful it would be.

**Review by Lyxal — APPROVED** [review]

File: shared/src/vyxal/ListHelpers.scala [hunk 4]
```diff
@@ -474,7 +474,11 @@ object ListHelpers:
     }
 
   def suffixes(iterable: VList): Seq[VList] =
-    prefixes(iterable.reverse).reverse.map(a => a.reverse)
+    LazyList.unfold(iterable) {
+      case remaining => Option.when(remaining.nonEmpty) {
```
[the remark this answers is not in the document]
Comment (Lyxal) [reply shared/src/vyxal/ListHelpers.scala]: Copilot 😜

File: shared/src/vyxal/Elements.scala [hunk 5]
```diff
@@ -1475,6 +1475,22 @@ object Elements:
...
     },
+    addPart(
+      Monad,
+      "Ṣ",
+      "Sublists",
+      List("sublists"),
+      false,
+      "a: lst -> sublists of a",
+    ) {
+      case a: (VVal | VList) => VList.from(
```
[the remark this answers is not in the document]
Comment (nayakrujul) [reply shared/src/vyxal/Elements.scala:1486]: Really? This is the method used in v2: https://github.com/Vyxal/Vyxal/blob/main/vyxal/elements.py#L6517

**Pull request merged**